data/api/client-server/capabilities.yaml

```
@@ -73,11 +73,17 @@ paths:
                           - default
                           - available
                       m.set_displayname:
+                        deprecated: true
                         $ref: '#/components/schemas/booleanCapability'
-                        description: Capability to indicate if the user can change their display name.
+                        description: |
+                          **Deprecated:** Capability to indicate if the user can change their display name.
+                          Refer to `m.profile_fields` for extended profile management.
                       m.set_avatar_url:
+                        deprecated: true
                         $ref: '#/components/schemas/booleanCapability'
-                        description: Capability to indicate if the user can change their avatar.
+                        description: |
+                          **Deprecated:** Capability to indicate if the user can change their avatar.
+                          Refer to `m.profile_fields` for extended profile management.
```

```suggestion
                        description: |
                          **Deprecated:** Capability to indicate if the user can change their avatar.
                          Refer to `m.profile_fields` for extended profile management.

                          For backwards compatibility, servers that directly or indirectly include the
                          `avatar_url` profile field in the `m.profile_fields` capability MUST also
                          set this capability accordingly.
```

Similar to https://github.com/matrix-org/matrix-spec/pull/2071/files#r2241865574.